### ISSUE TITLE
omni.kit.livestream.webrtc bundles an outdated libcudart.so.12 which breaks PyTorch import

### ISSUE THREAD

**Issue by vgerber** [isaac-sim/IsaacSim#747]
### Description

Hi,

we are building custom kit apps and include the webrtc extension as well. Our extension is using a pytorch module, but it crashed with a symbol missing (see nm command down below). We only manged to fix it, by patching the webrtc extension with the right binary. It looks like all webrtc extension are shipping this invalid binary.

### Isaac Sim version

110.2

### Operating System (OS)

Ubuntu 24.04

### GPU Name

RTX 5090 / 6000

### GPU Driver and CUDA versions

580

### Logs

```
[Error] Import error: .../omni.isaac.ml_archive-3.3.0+110.1.2.lx64.cp312/pip_prebundle/torch/lib/libc10_cuda.so:
undefined symbol: cudaGetDriverEntryPointByVersion, version libcuda
```

### Additional information

omni.kit.livestream.webrtc ships its own CUDA runtime at bin/libcudart.so.12. In every build we inspected — 10.3.2+110.0.0 (currently shipped inside Isaac Sim 6.0.1) up to 10.7.0+110.1.2 (newest linux-x86_64 build we could find) — this copy predates CUDA 12.5 and does not export cudaGetDriverEntryPointByVersion.

When the streaming extension loads before any extension that needs a modern CUDA 12.x runtime (e.g. omni.isaac.ml_archive / PyTorch), the stale copy claims the libcudart.so.12 soname in the process. Any later import torch then

Reproduction

1. Build a Kit streaming app (e.g. from the omni.streaming_configuration template) so omni.kit.livestream.webrtc starts at app boot.
2. After startup, enable any extension that imports torch from omni.isaac.ml_archive (e.g. via isaacsim.core.*).
3. Torch import fails with the undefined-symbol error above.

Isaac Sim itself does not hit this only because its torch-side exte.so.12 at boot, before streaming starts — i.e. the bug is masked byload order, not fixed.

Verification

> $ nm -D --with-symbol-versions omni.kit.livestream.webrtc-10.7.0+110.1.2.lx64.r.cp312/bin/libcudart.so.12 | grep -c cudaGetDriverEntryPointByVersion

The libcudart.so.12 shipped in current omni.gpu_foundation builds ethe copy inside the webrtc extension with it resolves the issue withno regression in streaming (CUDA 12 runtimes are backward compatible).

Expected behavior

omni.kit.livestream.webrtc should bundle a current CUDA 12.x runtim all and rely on the one provided by omni.gpu_foundation), so thatprocess-wide libcudart.so.12 resolution never lands on a runtime older than what other stock extensions (e.g. the torch prebundle) require.

Environment

- Kit SDK: 110.2.0 (also reproducible against 110.1.x packages)
- omni.kit.livestream.webrtc: 10.3.2+110.0.0 and 10.7.0+110.1.2 (both affected)
- omni.isaac.ml_archive: 3.3.0+110.1.2 (torch built against CUDA ≥
- OS: Ubuntu 24.04, linux-x86_64

```
Thu Aug 13 17:52:20 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 580.159.03             Driver Version: 580.159.03     CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 5090        Off |   00000000:01:00.0  On |                  N/A |
|  0%   53C    P1             77W /  600W |    5172MiB /  32607MiB |      5%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            2573      G   /usr/lib/xorg/Xorg                       20MiB |
|    0   N/A  N/A          853614      G   /usr/lib/xorg/Xorg                      166MiB |
|    0   N/A  N/A          853801      G   /usr/bin/gnome-shell                     33MiB |
|    0   N/A  N/A          854453      C   /usr/NX/bin/nxnode.bin                  535MiB |
|    0   N/A  N/A          854782      G   /usr/bin/nautilus                        19MiB |
|    0   N/A  N/A         1056638    C+G   .../linux-x86_64/release/kit/kit       4259MiB |
+-----------------------------------------------------------------------------------------+
```

**Comment by PeterL-NV** [comment]
Hi, thanks for reporting the issue. Looks like this needs to be addressed in Kit, will follow up.

**Comment by PeterL-NV** [comment]
Internal ticket 6607016

**Comment by PeterL-NV** [comment]
This is fixed in Kit 110.4 and will be in next Isaac Sim release. Closing.
